src/components/slide-toggle/slide-toggle.ts

```diff
@@ -58,6 +59,13 @@ export class MdSlideToggle implements AfterContentInit, ControlValueAccessor {
   private _hasFocus: boolean = false;
   private _isMousedown: boolean = false;
   private _isInitialized: boolean = false;
+  private _slideRenderer: MdSlideToggleRenderer = null;
+
+  // State of the current drag, which holds required variables for the drag.
+  private _dragState: {
+    barWidth: number;
+    percentage?: number;
+  };

```

Comment:
What do you think of moving the dragState values to be properties of the SliderRenderer? The `updateThumbPosition` function would change to take the `deltaX` from the event. Your event handlers could then look something like
```ts
_onSlideStart() {
  this._slideRenderer.startSliding();
}

_onSlide(slideEvent: HammerInput) {
  this._slideRenderer.updateThumbPosition(event.deltaX);
}

_onSlideEnd() {
  this._slideRenderer.stopSliding();
}
```

and the renderer could look something like
```ts
class SlideRenderer {
  private _thumb: HTMLElement;
  private _slideBar: HTMLElement;
  private _slideBarWidth: number;

  constructor(elementRef: ElementRef) {
    this._thumb = elementRef.nativeElement.querySelector('.md-slide-toggle-thumb-container');
    this._slideBar = elementRef.nativeElement.querySelector('.md-slide-toggle-bar');
  }

  isSliding(): boolean {
    return !!this._slideBarWidth;
  }

  startSliding(): void {
    if (!this.isSliding()) {
      // We can't grab the width until we actually start sliding in order to work with server-side pre-rendering.
      // Assume that the width of the slide bar doesn't change during one slide.
      this._sldieBarWidth = this._slideBar.clientWidth - this._thumb.clientWidth;
    }
  }

  stopSliding(): void {
    if (this.isSliding()) {
      this._slideBarWidth = null;
      this._clearThumbPosition();
    }
  }

  private _clearThumbPosition(): void {
    applyCssTransform(this._thumb, '');
  }

  updateThumbPosition(deltaX: number): void {
    let transformPercent = this._getThumbTransformPercent(deltaX);
    applyCssTransform(this._thumb, `translate3d(${transformPercent}%, 0, 0)`);
  }

  private _getThumbTransformPercent(deltaX: number): void {
    let percentage = (event.deltaX / this._dragState.barWidth) * 100;
    // When the slide-toggle was initially checked, then we have to start the drag at 100%
    if (this.checked) {
      percentage += 100;
    }

    return Math.max(0, Math.min(percentage, 100));    
  }
}
```
...okay, got a little carried away, but I think this is good :smiley: . (I wrote it in GitHub, so there might be some typos). 

Notes on the above:
* All of the sliding logic is encapsulated in the slide renderer.
* We only `querySelector` once per slider instance, since the elements won't ever change. 
* `clientWidth` should do the same job as `getBoundingClientRect().width` in this case
* Function names are more straightforward and the functions are more single-responsibility
* I left out the comments for brevity's sake in this comment